docs_src/design/adr/security/0019-spiffe.md

```diff
@@ -0,0 +1,338 @@
+  Now, with the added compilication of Consul tokens,
+  the EdgeX framework must be shut down and restarted completely
+  in order to change the list of running services.
+  It is not possible to delay the start of EdgeX services until a later time
+  (that is, greater than the default token TTL),
+  as they will not be able to connect to the EdgeX secret store
+  to obtain required secrets.
+
+* Transmission of the authentication token requires one or more shared file systems
+  between the service and `security-secretstore-setup`.
+  In the Docker implementation,
+  this shared file system is constructed by bind-mounting a host-based directory
+  to multiple containers.
+  The snap implementation is similar, utilizing a content-interface between snaps.
+  Similar functionality is near-impossible in a Kubernetes environment
+  that supports only RWO volumes:
+  the volume containing the secret can be mounted to only a single pod at a time.
+  It is simply easier to run without security enabled in Kubernetes.
+
+* The current approach cannot support distributed services
+  without an underlying distributed file system to distribute tokens,
+  such as GlusterFS, running across the participating nodes.
+
+## Decision
+
+EdgeX will create a new service, `security-spiffe-token-provider`.
+This service will be a mutual-auth TLS service that exchanges
+a [SPIFFE](https://spiffe.io/) X.509 SVID for a secret store token.
+
+The SPIFFE token provider will take theee parameters:
+
+1. An X.509 SVID used in mutual-auth TLS for the token provider
+   and the service to cross-authenticate.
+   
+2. The reqested service key.  If blank, the service key will
+   default to the service name encoded in the SVID.
+   If the service name follows the pattern `device-(name)`,
+   then the service key must follow the format
+   `device-(name)` or `device-name-*`.
+   If the service name is `app-service-configurable`,
+   then the service key must follow the format
+   `app-*`.
+
+3. A list of "known secret" identifiers that will allow
+   new services to request database passwords or
+   other "known secrets" to be seeded into their service's
+   partition in the secret store.
+
+The `go-mod-secrets` module will be modified to enable a new mode
+whereby a secret store token is obtained by:
+
+1. Obtaining an X.509 SVID by contacting a local SPIFFE agent's
+   workload API on a local Unix domain socket.
+2. Connecting to the `security-spiffe-token-provider` service
+   and using the X.509 SVID and request a secret store token.
+
+The SPIFFE authentication mode will be an opt-in feature.
+
+This feature is estimated to be a "large" or "extra large"
+effort that could be implemented in a single release cycle.
+
+### Technical Architecture
+
+![SPIFFE Architecture and Workflow](0019-spiffe-architecture.jpg)
+
+The work flow is as follows:
+
+1. The SPIFFE server is started.
+1. The server creates an initial set of keys.
```

Comment:
What kind of keys and what are they used for (I think I know the answer but we should call this out)?